### PR TITLE
86b1u133x

### PR DESCRIPTION
**Pull request by thlurte**
This PR tries to demonstrate a simple working example of role based routing and component rendering.
 
**Changes**
- Refactored the middleware logic to better handle user roles (student, mentor) and redirect them appropriately based on their roles.
- Resolved an issue in the `docker-config.yml` file.
- Introduced additional pages to demonstrate the role based routing and component rendering.

*Notes*: 
- No breaking changes are expected.

### CODE REVIEW

**Review by bsaranga — CHANGES_REQUESTED** [review]
Resolve conflicts

**Comment by thlurte** [comment]
> Resolve conflicts

the issues have been resolved

File: server/package.json [hunk 1]
```diff
@@ -19,20 +19,26 @@
...
-    "@prisma/client": "^5.19.0",
+    "@prisma/client": "^5.18.0",
     "@radix-ui/react-alert-dialog": "^1.1.1",
     "@radix-ui/react-dialog": "^1.1.1",
     "@radix-ui/react-slot": "^1.1.0",
     "@shadcn/ui": "^0.0.4",
+    "@types/bcryptjs": "^2.4.6",
+    "@types/cookie-parser": "^1.4.7",
+    "@types/express": "^4.17.21",
+    "@types/jsonwebtoken": "^9.0.6",
```
Comment (bsaranga) [inline server/package.json]: Remove these.

File: server/package.json [hunk 2]
```diff
@@ -54,6 +60,7 @@
     "jest": "^29.7.0",
     "postcss": "^8",
     "prisma": "^5.18.0",
+    "prisma": "^5.18.0",
```
Comment (bsaranga) [inline server/package.json]: Remove duplicate entry.

**Review by bsaranga — CHANGES_REQUESTED** [review]
Work on changed requested

File: server/package.json (hunk 2 above)
Comment (thlurte) [reply server/package.json]: i have removed them

**Review by bsaranga — APPROVED** [review]

**Pull request merged**